rfcs/0042-omnilock/0042-omnilock.md

```diff
@@ -0,0 +1,558 @@
+
+
+* 0x06: It follows the same unlocking method used by [CKB
+  MultiSig](https://github.com/nervosnetwork/ckb-system-scripts/blob/master/c/secp256k1_blake160_multisig_all.c)
+
+* 0xFC: The auth content that represents the blake160 hash of a lock script. The lock script will check if the current
+  transaction contains an input cell with a matching lock script. Otherwise, it would return with an error. It's similar
+  to [P2SH in BTC](https://en.bitcoin.it/wiki/Pay_to_script_hash).
+
+* 0xFD: The auth content that represents the blake160 hash of a preimage. The preimage contains
+  [exec](https://github.com/nervosnetwork/rfcs/pull/237) information that is used to delegate signature verification to
+  another script via exec.
+
+* 0xFE: The auth content that represents the blake160 hash of a preimage. The preimage contains [dynamic
+  linking](https://docs.nervos.org/docs/labs/capsule-dynamic-loading-tutorial/) information that is used to delegate
+  signature verification to the dynamic linking script. The interface described in [Swappable Signature Verification
+  Protocol Spec](https://talk.nervos.org/t/rfc-swappable-signature-verification-protocol-spec/4802) is used here.
+
+
+## Omnilock Script
+
+An Omnilock script has the following structure:
+```text
+Code hash: Omnilock script code hash
+Hash type: Omnilock script hash type
+Args: <21 byte auth> <Omnilock args>
+```
+
+among which, the structure of `<Omnilock args>` is as follows:
+
+```
+<1 byte Omnilock flags> <32 byte AdminList cell Type ID, optional> <2 bytes minimum ckb/udt in ACP, optional> <8 bytes since for time lock, optional> <32 bytes type script hash for supply, optional>
+```
+
+| Name               | Flags      | Affected Args   |Affected Args Size (byte)|Affected Witness
+| -------------------|------------|-----------------|---------|-------------------------------- 
+| administrator mode | 0b00000001 |	AdminList cell Type ID | 32      | omni_identity/signature in OmniLockWitnessLock
+| anyone-can-pay mode| 0b00000010 | minimum ckb/udt in ACP| 2 | N/A
+| time-lock mode     | 0b00000100 | since for timelock| 8     | N/A
+| supply mode        | 0b00001000 |type script hash for supply| 32 | N/A
+| auth               | N/A        |21-byte auth identity | 21 | signature in OmniLockWitnessLock
+
+
+## Administrator Mode
+
+When "administrator mode" is enabled, `<32 byte AdminList cell Type ID>` must be present. The AdminList cell contains the
+type script hash used by a special cell with the same format as [RCE
+Cell](https://talk.nervos.org/t/rfc-regulation-compliance-extension/5338). The RCE cell follows a set of rules and contains
+whitelists and blacklists. These lists can be used in the [SMT proofs
+scenarios](https://github.com/nervosnetwork/sparse-merkle-tree).
+
+The RCE cells are organized in tree structure illustrated in the following diagram:
+
+![RCE Cells](./rce_cells.png)
```

Comment:
Please share the diagram source file in comment.

Reply:
https://drive.google.com/file/d/1mKgqIl2IK-n9JV0kktT2zZDvDaRsUS7D/view?usp=sharing
Open it with draw.io